core/src/main/java/com/sequenceiq/cloudbreak/service/securityconfig/SecurityConfigService.java

```diff
@@ -40,12 +42,17 @@ public Optional<SecurityConfig> findOneByStackId(Long stackId) {
         return securityConfigRepository.findOneByStackId(stackId);
     }
 
+    public SecurityConfig getOneByStackId(Long stackId) {
+        return findOneByStackId(stackId)
+                .orElseThrow(() -> new BadRequestException("Stack does not have a security config"));
```

Comment:
`NotFoundException` or `CloudbreakServiceException` would fit this case more, as it's not a request error